Source/PlasticSourceControl/Private/PlasticSourceControlOperations.cpp

```diff
@@ -1810,6 +1821,94 @@ bool FPlasticShelveWorker::UpdateStates()
 }
 
 
+
+FName FPlasticUnshelveWorker::GetName() const
+{
+	return "Unshelve";
+}
+
+bool FPlasticUnshelveWorker::Execute(FPlasticSourceControlCommand& InCommand)
+{
+	TRACE_CPUPROFILER_EVENT_SCOPE(FPlasticUnshelveWorker::Execute);
+
+	// Get the state of the changelist to operate on
+	TSharedRef<FPlasticSourceControlChangelistState, ESPMode::ThreadSafe> ChangelistState = GetProvider().GetStateInternal(InCommand.Changelist);
+
+	// Detect if any file to unshelve has some local modification, which would fail the "unshelve" operation with a merge conflict
+	// NOTE: we could decide to automatically undo the local changes in order for this process too be automatic, like with Perforce
+	for (const FString& File : InCommand.Files)
+	{
+		if (ChangelistState->Files.FindByPredicate([&File](const FSourceControlStateRef& FileState) { return FileState->GetFilename().Equals(File, ESearchCase::IgnoreCase); }))
+		{
+			UE_LOG(LogSourceControl, Error, TEXT("Revert local changes in order to unshelve the corresponding changes from the shelve."));
+			return false;
+		}
+	}
+
+	// Get the list of files to unshelve if not all of them are selected
+	TArray<FString> Files;
+	if (InCommand.Files.Num() < ChangelistState->ShelvedFiles.Num())
+	{
+		if (GetProvider().GetPlasticScmVersion() < PlasticSourceControlVersions::ShelvesetApplySelection)
+		{
+			// On old version, don't unshelve the files if they are not all selected (since we couldn't apply only a selection of files from a shelve)
+			UE_LOG(LogSourceControl, Error,
+				TEXT("Plastic SCM %s cannot unshelve a selection of files from a shelve. Unshelve them all at once or update to %s."),
```

Comment:
It might be worth asking users to update to plastic version `PlasticSourceControlVersions::ShelvesetApplySelection` **or above**.